Gems/PhysX/Code/Source/Scene/PhysXScene.cpp

```diff
@@ -1287,4 +1307,46 @@ namespace PhysX
         }
     }
 
-}
+    void PhysXScene::FlushTransformSync()
+    {
+        AZ_PROFILE_SCOPE(Physics, "PhysX::FlushTransformSync");
+
+        m_queuedActiveBodyIndices.Apply(
+            [this](AzPhysics::SimulatedBodyIndex bodyIndex)
+            {
+                if (bodyIndex < m_simulatedBodies.size() && m_simulatedBodies[bodyIndex].second)
+                {
+                    m_simulatedBodies[bodyIndex].second->SyncTransform(m_accumulatedDeltaTime);
+                }
+            });
+
+        m_queuedActiveBodyIndices.Clear();
+        m_accumulatedDeltaTime = 0.0f;
+    }
+
+    void PhysXScene::QueuedActiveBodyIndices::Insert(AzPhysics::SimulatedBodyIndex bodyIndex)
+    {
+        if (m_set.insert(bodyIndex).second)
+        {
+            m_array.emplace_back(bodyIndex);
+        }
+    }
+
+    void PhysXScene::QueuedActiveBodyIndices::Reserve(size_t reserveSize)
+    {
+        m_array.reserve(m_array.size() + reserveSize);
+    }
```

Comment:
This function is doing something different than a `reserve`, it's always increasing memory of `m_array`. Why `m_array.reserve(reserveSize);` wouldn't work? 

If always increasing memory is what's needed, then maybe use another name that reflects better what's doing, for example `IncreaseCapacity` or something on those lines which is clear it's always incrementing.